libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/view/MapboxSecondaryManeuver.kt

```diff
@@ -0,0 +1,155 @@
+package com.mapbox.navigation.ui.maneuver.view
+
+import android.content.Context
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.Drawable
+import android.text.SpannableStringBuilder
+import android.text.Spanned
+import android.text.style.ImageSpan
+import android.util.AttributeSet
+import androidx.appcompat.widget.AppCompatTextView
+import androidx.core.content.ContextCompat
+import com.mapbox.navigation.ui.base.MapboxView
+import com.mapbox.navigation.ui.base.model.maneuver.DelimiterComponentNode
+import com.mapbox.navigation.ui.base.model.maneuver.ExitNumberComponentNode
+import com.mapbox.navigation.ui.base.model.maneuver.ManeuverState
+import com.mapbox.navigation.ui.base.model.maneuver.RoadShieldComponentNode
+import com.mapbox.navigation.ui.base.model.maneuver.SecondaryManeuver
+import com.mapbox.navigation.ui.base.model.maneuver.TextComponentNode
+import com.mapbox.navigation.ui.maneuver.R
+
+/**
+ * Default view to render secondary banner instructions onto [MapboxManeuverView].
+ * It can be directly used in any other layout.
+ * @property attrs AttributeSet
+ * @property defStyleAttr Int
+ */
+class MapboxSecondaryManeuver @JvmOverloads constructor(
```

Comment:
There's a lot of duplicate code between primary/secondary/sub maneuvers. Could you extract those utils to separate classes and reuse them?

Reply:
I agree, but I already tried. The code in these files are very UI specific, so they need access to components like `Context`, `Drawables`, `Bitmaps`, `TextView` etc and it gets complicated. Moving them to `Utility` will make it difficult to test and maintain.

Reply:
I'm sure we can design static methods, maybe with a longer arguments list, to process the data. Now we have copy-pasted code - we have to test it twice and if there's a bug in it, we'll have to fix it twice.